### PR TITLE
Adding SUST source terms to SST, default to off

### PR DESCRIPTION
**Pull request by marchdf**
**Pull-request type:**
- [ ] Bug fix 
- [x] Documentation update
- [X] Feature enhancement

## Description of the pull-request

This adds the capability to add the SUST source terms to SST (described at [the NASA LARC website](https://turbmodels.larc.nasa.gov/sst.html)). The user gets to pick two more turbulence parameters, `tke_amb` and `sdr_amb`, that set the freestream values. An appropriate source term is then added to the SST equations. The constants default to 0.0 to preserve the current behavior. 

This capability is necessary for external aerodynamic flows to avoid the drastic decay of SDR and TKE from the inflow BC to the airfoil.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [x] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [x] Linux
    - [ ] MacOS
  - Compilers 
    - [x] GCC
    - [ ] LLVM/Clang
    - [ ] Intel compilers
    - [ ] NVIDIA CUDA
- [x] Compiles without warnings
- [x] Passes all unit tests
- [ ] Passes all regression tests
- [x] Documentation builds without errors

*For new code updates*
- [x] Documentation updates - additions to user, theory, and verification manuals
- [ ] New unit tests providing coverage for new code, bug fixes etc.
- [x] New regression tests exercising the new code

### CODE REVIEW

**Comment by marchdf** [comment]
This is a draft until I complete the documentation update

**Comment by marchdf** [comment]
Just as an illustration. This PR fixes some strange transients that show up in SDR and F1 blending when using the baseline SST:

SDR: left: SST with SUST, right: baseline
Screen Shot 2022-03-04 at 9 30 21 AM

F1 blending function
Screen Shot 2022-03-04 at 9 28 53 AM

But the flow in general reaches a similar state:
Screen Shot 2022-03-04 at 9 34 02 AM

with minor differences in the forces:

Baseline:
```
13.3333          3.4575         194.493               0        0.909228       0.0724233               0               0               0         10.4706       0.0407199         3.87341
```

with SUST:
```
13.3333         3.59349         191.424               0         0.90253       0.0717434               0               0               0          10.237       0.0395102         3.90931
```

**Review by psakievich — APPROVED** [review]

**Review by jamelvin — APPROVED** [review]
Looks good

File: include/Enums.h [hunk 1]
```diff
@@ -234,12 +234,14 @@ enum TurbulenceModelConstant {
...
   TM_ams_peclet_scale = 52,
-  TM_caOne = 53,
-  TM_caTwo = 54,
-  TM_ceOne = 55,
-  TM_ceTwo = 56,
-  TM_c0t = 57,
-  TM_END = 58
+  TM_tkeAmb = 53,
+  TM_sdrAmb = 54,
+  TM_caOne = 55,
```
[the remark this answers is not in the document]
Comment (marchdf) [reply include/Enums.h:239]: Yeah I noticed that too but I figured it was coming in the future PRs that @wilsonrvornlgov is making.

[the remark this answers is not in the document]
Comment (wilsonrvornlgov) [reply include/Enums.h:239]: It looks like they are set in SolutionOptions.C:

  turbModelConstantMap_[TM_caOne] = 2.0;
  turbModelConstantMap_[TM_caTwo] = 0.06;
  turbModelConstantMap_[TM_ceOne] = 1.0;
  turbModelConstantMap_[TM_ceTwo] = 50.0;
  turbModelConstantMap_[TM_c0t] = 0.03;

[the remark this answers is not in the document]
Comment (jamelvin) [reply include/Enums.h:239]: Hey @wilsonrvornlgov, yeah I think you set everything up for them to function as you wanted without being a true "input option".  What @marchdf and I were referring to was there didn't seem to be a corresponding text entry in the names list for the turbulence model constants in Enums.h, see 

https://github.com/Exawind/nalu-wind/blob/cbfddf9794f397252eeb289f8b8ff23ef8dfcaca/include/Enums.h#L247 .

This inconsistency between those two lists could cause issues down the road for anyone adding to the list that wants a future turbulence model constant to be modifiable in the input file.

[the remark this answers is not in the document]
Comment (marchdf) [reply include/Enums.h:239]: Adding them to the list Jeremy is referring to will make them able to be set in the input file.

[the remark this answers is not in the document]
Comment (wilsonrvornlgov) [reply include/Enums.h:239]: Thanks for the suggestion. I will add that change to the next MR.

**Pull request merged**